test/lib/ansible_test/_util/controller/sanity/pylint/plugins/deprecated.py

```diff
@@ -277,6 +279,62 @@ def visit_call(self, node):
             pass
 
 
+class AnsibleDeprecatedCommentChecker(BaseTokenChecker):
+    """Checks for ``# deprecated: X.Y`` comments to ensure that the ``version``
+    has not passed or met the time for removal
+    """
+
+    __implements__ = (ITokenChecker,)
+
+    name = 'deprecated-comment'
+    msgs = {
+        'E9601': ("Deprecated core version (%r) found: %s",
+                  "ansible-deprecated-version-comment",
+                  "Used when a '# deprecated:' comment specifies a version "
+                  "less than or equal to the current version of Ansible",
+                  {'minversion': (2, 6)}),
+        'E9602': ("Deprecated comment contains invalid keys %r",
+                  "ansible-deprecated-version-comment-invalid-key",
+                  "Used when a '#deprecated:' comment specifies invalid data",
+                  {'minversion': (2, 6)}),
+    }
+
+    def process_tokens(self, tokens: list[TokenInfo]) -> None:
+        for token in tokens:
+            if token.type == COMMENT:
+                self._process_comment(token)
+
+    def _deprecated_string_to_dict(self, token: TokenInfo, string: str) -> dict[str, str]:
+        data = {'description': None, 'core_version': None}
```

Comment:
Is `core_version` intentionally optional? It seems like we'd want to require it.

Reply:
There is a "plan" to allow `python_version` as an alternative to `core_version`. We don't have the feasibility right now necessarily to do that, but that's why I left it optional, as we want to extend this to be more specific as to the version necessitating the removal.

Reply:
We already have the supported Python version information in ansible-test, which could be exposed to the plugin. If we're not using that now, it seems like we'd want to make `core_version` required until `python_version` is available (then require one of them).

Reply:
> We already have the supported Python version information in ansible-test, which could be exposed to the plugin. 

If you have a recommendation on doing so, I can try to implement that.

For now, I've just made `core_version` required.

Reply:
Ok, I've added in `python_version` support, dunno if you had other thoughts, but just let me know.  I'll mark this is resolved.

Reply:
It's close to what I was thinking. Do you mind if I push a commit to adjust the path filtering for deciding if a controller or target Python should be used?

Reply:
@mattclay absolutely, feel free.